### PR TITLE
Added missing dependency.

### PR DESCRIPTION
**Pull request by nicksantamaria**
Resolves an issue when installing a `tide` profile from scratch.

```
Drupal\Core\Config\UnmetDependenciesException: Configuration objects provided by <em                                     [error]
class="placeholder">tide_landing_page</em> have unmet dependencies: <em
class="placeholder">field.field.paragraph.form_embed_openforms.field_form_link (link_field_autocomplete_filter)</em>
in /app/docroot/core/lib/Drupal/Core/Config/UnmetDependenciesException.php:98
Stack trace:
#0 /app/docroot/core/lib/Drupal/Core/Config/ConfigInstaller.php(509):
Drupal\Core\Config\UnmetDependenciesException::create('tide_landing_pa...', Array)
#1 /app/docroot/core/lib/Drupal/Core/ProxyClass/Config/ConfigInstaller.php(132):
Drupal\Core\Config\ConfigInstaller->checkConfigurationToInstall('module', 'tide_landing_pa...')
#2 /app/docroot/core/lib/Drupal/Core/Extension/ModuleInstaller.php(161):
Drupal\Core\ProxyClass\Config\ConfigInstaller->checkConfigurationToInstall('module', 'tide_landing_pa...')
#3 /app/docroot/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83):
Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#4 /app/docroot/core/includes/install.core.inc(1877):
Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array, false)
#5 /app/docroot/core/includes/batch.inc(295): _install_module_batch('tide_landing_pa...', 'Tide Landing Pa...',
Array)
#6 /app/docroot/core/includes/form.inc(947): _batch_process()
#7 /app/docroot/modules/contrib/tide_core/tide_core.install(338): batch_process()
#8 /app/docroot/modules/contrib/tide_core/tide_core.install(67): tide_core_update_8009()
#9 [internal function]: tide_core_install()
#10 /app/docroot/core/lib/Drupal/Core/Extension/ModuleHandler.php(392): call_user_func_array('tide_core_insta...',
Array)
#11 /app/docroot/core/lib/Drupal/Core/Extension/ModuleInstaller.php(326):
Drupal\Core\Extension\ModuleHandler->invoke('tide_core', 'install')
#12 /app/docroot/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83):
Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#13 /app/docroot/core/includes/install.core.inc(1877):
Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array, false)
#14 /app/docroot/core/includes/batch.inc(295): _install_module_batch('tide_core', 'Tide Core', Array)
#15 /app/docroot/core/includes/form.inc(947): _batch_process()
#16 /app/docroot/core/includes/install.core.inc(659): batch_process(Object(Drupal\Core\Url), Object(Drupal\Core\Url))
#17 /app/docroot/core/includes/install.core.inc(577): install_run_task(Array, Array)
#18 /app/docroot/core/includes/install.core.inc(117): install_run_tasks(Array, NULL)
#19 /app/vendor/drush/drush/includes/drush.inc(728): install_drupal(Object(Composer\Autoload\ClassLoader), Array)
#20 /app/vendor/drush/drush/includes/drush.inc(713): drush_call_user_func_array('install_drupal', Array)
#21 /app/vendor/drush/drush/commands/core/drupal/site_install.inc(82): drush_op('install_drupal',
Object(Composer\Autoload\ClassLoader), Array)
#22 /app/vendor/drush/drush/commands/core/site_install.drush.inc(271): drush_core_site_install_version('tide', Array)
#23 /app/vendor/drush/drush/includes/command.inc(422): drush_core_site_install('tide', 'install_configu...',
'install_configu...')
#24 /app/vendor/drush/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#25 /app/vendor/drush/drush/includes/command.inc(199): drush_command('tide', 'install_configu...',
'install_configu...')
#26 /app/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#27 /app/vendor/drush/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#28 phar:///usr/local/bin/drush/bin/drush.php(153): drush_main()
#29 /usr/local/bin/drush(10): require('phar:///usr/loc...')
#30 {main}
```

### CODE REVIEW

**Comment by vincent-gao** [comment]
emm `dpc-sdp/tide_core` has `drupal/link_field_autocomplete_filter` dependency, and `tide_landing_page` requires `tide_core`~

**Review by vincent-gao — APPROVED** [review]

**Comment by sonnykt** [comment]
> emm `dpc-sdp/tide_core` has `drupal/link_field_autocomplete_filter` dependency, and `tide_landing_page` requires `tide_core`~

@vincent-gao `tide_core` doesn't install `link_field_autocomplete_filter`. It has `link_field_autocomplete_filter` as a composer dependency, but not Drupal module dependency.

**Comment by nicksantamaria** [comment]
So does this change need to move into `tide_core`?

**Comment by sonnykt** [comment]
> So does this change need to move into `tide_core`? There is some link_field_autocomplete_filter config in this module. I think it should remain as an explicit dependency here as well.
> 
> https://github.com/dpc-sdp/tide_landing_page/blob/97019116d3a9e56044fd20050cf50e663b91bc68/config/install/field.field.paragraph.form_embed_openforms.field_form_link.yml#L11

either tide_core or tide_landing_page is fine. Landing page is the only content type using that feature so far. Enabling it in tide_core allows other modules to use it as well.

File: tide_landing_page.info.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies:
   - maxlength:maxlength
   - paragraphs:paragraphs
   - smart_trim:smart_trim
+  - link_field_autocomplete_filter
```
Comment (sonnykt) [inline tide_landing_page.info.yml:26]:
```suggestion
  - link_field_autocomplete_filter: link_field_autocomplete_filter
```

**Review by sonnykt — CHANGES_REQUESTED** [review]

**Comment by anthony-malkoun** [comment]
Going to close this in favour of moving the dependency to `tide_core`. See branch `feature/DDS-809-implement-scaffold`

**Pull request closed without merging**